src/resnext.jl

```diff
@@ -0,0 +1,117 @@
+
+# Arguments
+- `cardinality`: the number of groups to use for the convolution
+- `width`: the number of feature maps in each group in the bottleneck
+- `connection`: the binary function applied to the output of residual and skip paths in a block
+- `block_config`: a list of the number of residual blocks at each stage
+- `nclasses`: the number of output classes
+"""
+function resnext(cardinality, width, widen_factor = 2, connection = (x, y) -> @. relu(x) + relu(y);
+    block_config, nclasses = 1000)
+    inplanes = 64
+    baseplanes = 128
+    layers = []
+    append!(layers, conv_bn((7, 7), 3, inplanes; stride = 2, pad = (3, 3)))
+    push!(layers, MaxPool((3, 3), stride = (2, 2), pad = (1, 1)))
+    for (i, nrepeats) in enumerate(block_config)
+        # output planes within a block
+        outplanes = baseplanes * widen_factor
+        # push first skip connection on using first residual
+        # downsample the residual path if this is the first repetition of a block
+        push!(layers, Parallel(connection, resnextblock(inplanes, outplanes, cardinality, width, widen_factor, i != 1),
+            skip_projection(inplanes, outplanes, i != 1)))
+        # push remaining skip connections on using second residual
+        inplanes = outplanes
+        for _ in 2:nrepeats
+            push!(layers, Parallel(connection, resnextblock(inplanes, outplanes, cardinality, width, widen_factor, false),
+                skip_identity(inplanes, outplanes, false)))
+        end
+        baseplanes = outplanes
+    end
+
+    return Chain(Chain(layers...),
+        Chain(AdaptiveMeanPool((1, 1)), flatten, Dense(inplanes, nclasses)))
+end
+
+"""
+    ResNeXt(cardinality, width; block_config, nclasses = 1000)
+    
+Create a ResNeXt model
+([reference](https://arxiv.org/abs/1611.05431)).
+
+# Arguments
+- `cardinality`: the number of groups to use for the convolution
+- `width`: the number of feature maps in each group in the bottleneck
+- `nclasses`: the number of output classes
+"""
+struct ResNeXt
+    layers
+end
+
+function ResNeXt(cardinality, width; block_config, nclasses = 1000)
+    layers = resnext(cardinality, width; block_config, nclasses)
+    ResNeXt(layers)
+end
+
+@functor ResNeXt
+
+(m::ResNeXt)(x) = m.layers(x)
+
+backbone(m::ResNeXt) = m.layers[1]
+classifier(m::ResNeXt) = m.layers[2]
+
+const resnext_config = Dict(
+    50 => (3, 4, 6, 3),
+    101 => (3, 4, 23, 3)
+)
+
+"""
+    ResNeXt(; config::Int = 50, cardinality = 32, width = 4, pretrain = false, nclasses = 1000)
+
+Create a ResNeXt model with specified configuration. Currently supported values are (50, 101).
+([reference](https://arxiv.org/abs/1611.05431)).
+Set `pretrain = true` to load the model with pre-trained weights for ImageNet.
+
+!!! warning
+    `ResNeXt` does not currently support pretrained weights.
+
+See also [`Metalhead.resnext`](#).
+"""
+function ResNeXt(; config::Int = 50, cardinality = 32, width = 4, pretrain = false, nclasses = 1000)
```

Comment:
`config` should be a positional argument, and `cardinality` and `width` should be part of the configuration dictionary.

Reply:
This one is tricky, because what the original ResNeXt paper allows for is changes to cardinality and width even while maintaining a single configuration of layers - they've given versions that are 64x4d, for instance. I wanted to allow for that. I'll change config to be a positional argument, though.